### PR TITLE
Bus leds

### PR DESCRIPTION
**Pull request by domiko96**

### CODE REVIEW

File: software/lib/AnimationHandlerBus/AdressableModes/noise.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        debugFkt("Frequency was set to 0 or over 60 this is not possible", ERROR);
+        return;
+    }
+
+    unsigned long now = millis();
+    static unsigned long lastStateChange = 0;
+    long period = (1000 / this->silo->frequency);
+
+    if (now >= lastStateChange + period)
```
Comment (domiko96) [inline software/lib/AnimationHandlerBus/AdressableModes/noise.cpp:18]: horny Shit dome

File: software/lib/AnimationHandlerBus/AnimationHandlerBus.cpp [hunk 2]
```diff
@@ -1,18 +1,101 @@
...
+        break;
+    case INDUSTRIAL_FLICKER:
+        return industrialFlicker();
+        break;
+    default:
+        debugFkt("The Selected Mode is not a uniform color mode (single color strip mode)", ERROR);
+        return CRGBWW{0, 0, 0, 0, 0};
+        break;
+    }
+}
```
Comment (domiko96) [inline software/lib/AnimationHandlerBus/AnimationHandlerBus.cpp:88]: Maybe move this in Superclass

File: software/lib/Definitions/TypeDefinitions.h [hunk 3]
```diff
@@ -2,25 +2,27 @@
...
-    }
-};
+class CRGBWW
+{
+public:
+  unsigned int R, G, B, WW, CW;
 
+  //operator overloading
+  //todo put this into an c++ file
+  bool operator==(const CRGBWW &color) const
```
Comment (domiko96) [inline software/lib/Definitions/TypeDefinitions.h:12]: Great Idea, should implement other operators as well

File: software/src/initFunctions.h [hunk 4]
```diff
@@ -251,7 +256,10 @@ void initStrip(){
   }
   else{
     //TODO
-    //busHandler = new AnimationHandlerBus()
+
+    //get strip length:
+    int length = atoi(strip_length);
+    busHandler = new AnimationHandlerBus(length, Silo, varSiloChanged, micHandler);
```
Comment (domiko96) [inline software/src/initFunctions.h:262]: This is ugly and wrong

File: software/lib/Definitions/TypeDefinitions.h (hunk 3 above)
Comment (aewatt95) [reply software/lib/Definitions/TypeDefinitions.h:12]: Like... you know =

File: software/src/initFunctions.h (hunk 4 above)
Comment (aewatt95) [reply software/src/initFunctions.h:262]: And by far the worst thing i've ever seen in my entire coding career

**Comment by aewatt95** [comment]
Good, but not good enough. Please try better the next time.

**Pull request closed without merging**